snuba/stateful_consumer/__init__.py

```diff
@@ -18,11 +22,27 @@ class ConsumerStateData:
     state to the other.
     """
     snapshot_id: Optional[str]
+    transaction_data: Optional[TransactionData]
 
     @classmethod
-    def no_snapshot_state(cls):
+    def no_snapshot_state(cls) -> ConsumerStateData:
         """
         Builds an empty ConsumerStateData that represent a state where there is no
         snapshot to care about.
         """
-        return ConsumerStateData(None)
+        return ConsumerStateData(None, None)
```

Comment:
I feel like we may have already talked about this before, but if there's a reason why this can't be represented as a `None` wherever it's used instead of a `ConsumerStateData` with no fields set, it should probably be commented. Ideally, wherever this is being used (is it just `bootstrap.py`?) could use an `Union[ConsumerStateData, None]` and both of these fields could be required (unless there's a case for only partially filling them?)

Reply:
I think the debate was whether to explicitly create the ConsumerStateData  instead of using a factory method. 
I am ok using None instead of an empty ConsumerStateData